op-challenger/fault/factory.go

```diff
@@ -0,0 +1,64 @@
+package fault
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// MinimalDisputeGameFactoryCaller is a minimal interface around [bindings.DisputeGameFactoryCaller].
+// This needs to be updated if the [bindings.DisputeGameFactoryCaller] interface changes.
+type MinimalDisputeGameFactoryCaller interface {
+	GameCount(opts *bind.CallOpts) (*big.Int, error)
+	GameAtIndex(opts *bind.CallOpts, _index *big.Int) (struct {
+		Proxy     common.Address
+		Timestamp *big.Int
+	}, error)
+}
+
+type FaultDisputeGame struct {
+	Proxy     common.Address
+	Timestamp *big.Int
+}
+
+// GameLoader is a minimal interface for fetching on chain dispute games.
+type GameLoader interface {
+	FetchAllGames(ctx context.Context) ([]FaultDisputeGame, error)
+}
+
+type gameLoader struct {
+	caller MinimalDisputeGameFactoryCaller
+}
+
+// NewGameLoader creates a new services that can be used to fetch on chain dispute games.
+func NewGameLoader(caller MinimalDisputeGameFactoryCaller) *gameLoader {
+	return &gameLoader{
+		caller: caller,
+	}
+}
+
+// FetchAllGames fetches all dispute games from the factory.
+// todo(refcell): batch call this to give mr. rpc a break.
+func (l *gameLoader) FetchAllGames(ctx context.Context) ([]FaultDisputeGame, error) {
+	gameCount, err := l.caller.GameCount(&bind.CallOpts{Context: ctx})
```

Comment:
We need to have a consistent view of the chain when performing multiple calls so shouldn't just use the latest head in every call.